### DISCUSSION TITLE
a replace of "arguments.0" in NewExpression always adds extra brackets

### DISCUSSION THREAD

**Discussion by capaj** [babel/babel#13286] [Q&A] [answered]
I am trying to write a simple transformer which adds an extra argument to a new Expression. https://astexplorer.net/#/gist/7b07baf1fd088b0f6f61d93750f7eee3/65db00d1aa6f71c0ebf3a6179b3d6eaa69fe5a34

THe problem is it always adds extra brackets around the replaced thing. I have tried replacing it with other methods too. Does anyone has a clue why it always adds extra brackets?

**Comment by JLHwung** [comment, marked as answer]
I presume you mean parentheses for brackets. The parentheses are added because Babel interprets the input as a sequence expression like `(a, b)`, so Babel replaces the first argument as-is.

If you want to _add_ a new node to `path.get("arguments")`, you should use the `pushContainer` method: https://astexplorer.net/#/gist/7b07baf1fd088b0f6f61d93750f7eee3/bea56984eee524ecddddd56eb16104c8f3600517

In this example I also replaced the `replaceSourceWithString` by Babel templates, because `replaceSourceWithString`  is very hacky and should be avoided if you are writing new plugins.
```js
export default function (babel) {
  const { types: t, template } = babel;
  
  return {
    name: "ast-transform", // not required
    visitor: {
         NewExpression(path) {
      if (path.parent.type === 'ThrowStatement') {
        // @ts-expect-error
        if (path.node.callee.name === 'Error') {
          return
        }
        const [firstArgument] = path.node.arguments
        if (firstArgument) {
          const line = path.parent.loc.start.line
        }
        path.get('arguments.0').replaceWith(
           template.ast`'${firstArgument.extra ? firstArgument.extra.rawValue: firstArgument.quasis[0].value.cooked}'`);
        path.pushContainer('arguments', template.expression.ast`aaa`); // <- push `aaa` to become `arguments.1`
      }
    }
    }
  };
}
```
